ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```
@@ -0,0 +1,195 @@
+    public static final String QUERY_FIELD = "query";
+
+    public static final String TYPE = "SearchIndexTool";
+    private static final String DEFAULT_DESCRIPTION =
+        "Use this tool to search index with a query. You should pass in two parameters: index and query. Index is the index name and the query is an OpenSearch DSL query.";
+
+    private String name = TYPE;
+
+    private String description = DEFAULT_DESCRIPTION;
+
+    private Client client;
+
+    private NamedXContentRegistry xContentRegistry;
+
+    public static final Gson gson;
+
+    static {
+        gson = new Gson();
+    }
+
+    public SearchIndexTool(Client client, NamedXContentRegistry xContentRegistry) {
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public String getVersion() {
+        return null;
+    }
+
+    @Override
+    public boolean validate(Map<String, String> parameters) {
+        if (parameters == null || !parameters.containsKey(INPUT_FIELD)) {
+            return false;
+        }
+        String input = parameters.get(INPUT_FIELD);
+        return input != null;
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        try {
+            String input = parameters.get(INPUT_FIELD);
+            String index = "";
+            String query = "";
+            try {
+                // need this try catch block, because index, query field may not exist
+                JsonObject jsonObject = gson.fromJson(input, JsonObject.class);
+                index = jsonObject.get(INDEX_FIELD).getAsString();
+                query = jsonObject.get(QUERY_FIELD).toString();
+                query = "{\"query\": " + query + "}";
+            } catch (Exception e) {
+                // throw new IllegalArgumentException("wrong input");
```

It's not a good practice to ignore the exception catched, you can at least print some error logs. But this case we better remove this try...catch.

---

Some exception might be thrown from `index = jsonObject.get(INDEX_FIELD).getAsString();` I will catch this exception and throw it in the catch block.

---

With the `Optional` optimization, I believe this try...catch can be removed.

---

This try catch has been removed.